xunit-test/TranscoderTest/TranscoderTest.cpp

```diff
@@ -0,0 +1,247 @@
+    TranscoderTest::remoteCacheManager->start();
+
+    JBasicMarshaller<std::string> *km = new JBasicMarshaller<std::string>();
+    JBasicMarshaller<std::string> *vm = new JBasicMarshaller<std::string>();
+    RemoteCache<std::string, std::string>& cache = TranscoderTest::remoteCacheManager->getCache<std::string, std::string>(km,
+            &Marshaller<std::string>::destroy,
+            vm,
+            &Marshaller<std::string>::destroy,
+            "transcodingCache", true);
+
+    // Define a cache with a raw media type for the value
+    // and check that it returns value with the jboss marshaller preamble
+    DataFormat<std::string, std::string> df1;
+    df1.keyMediaType = MediaType(APPLICATION_JBOSS_MARSHALLING_TYPE);
+    df1.valueMediaType = MediaType(APPLICATION_JSON_TYPE);
+    df1.valueMarshaller.reset(new BasicMarshaller<std::string>());
+    RemoteCache<std::string, std::string> rawCache = cache.withDataFormat(&df1);
+
+    std::string k1("key13");
+    std::string v1("boron");
+
+    cache.clear();
+    cache.put(k1,v1);
+
+    std::shared_ptr<std::string> jv1(rawCache.get(k1));
+    // We got a clean json string
+    ASSERT_EQ(*jv1, "\"boron\"");
+
+    //Check that the initial cache works as usual
+    std::shared_ptr<std::string> result_v1(cache.get(k1));
+    ASSERT_EQ(*result_v1, "boron");
+}
+
+// Write jboss format and read raw json. Cache is jboss marshalled, so integers are recognized and transcoded correctly
+TEST_F(TranscoderTest, JBossMarshalledCacheWriteIntegerJbossReadJson) {
+    TranscoderTest::remoteCacheManager->start();
+
+    JBasicMarshaller<std::string> *km = new JBasicMarshaller<std::string>();
+    JBasicMarshaller<int> *vm = new JBasicMarshaller<int>();
+    RemoteCache<std::string, int>& cache = TranscoderTest::remoteCacheManager->getCache<std::string, int>(km,
+            &Marshaller<std::string>::destroy,
+            vm,
+            &Marshaller<int>::destroy,
+            "transcodingCache", true);
+
+    // Define a cache with a raw media type for the value
+    // and check that it returns value with the jboss marshaller preamble
+    DataFormat<std::string, std::string> df1;
+    df1.keyMediaType = MediaType(APPLICATION_JBOSS_MARSHALLING_TYPE);
+    df1.valueMediaType = MediaType(APPLICATION_JSON_TYPE);
+    df1.valueMarshaller.reset(new BasicMarshaller<std::string>());
+    df1.keyMarshaller.reset(new JBasicMarshaller<std::string>());
+    RemoteCache<std::string, std::string> rawCache = cache.withDataFormat(&df1);
+
+    std::string k1("key13");
+    int v1(42);
+
+    cache.clear();
+    cache.put(k1,v1);
+
+    std::shared_ptr<std::string> jv1(rawCache.get(k1));
+    // We got a clean json integer
+    ASSERT_EQ(*jv1, "42");
+
+    //Check that the initial cache works as usual
+    std::shared_ptr<int> result_v1(cache.get(k1));
+    ASSERT_EQ(*result_v1, 42);
+}
+
+TEST_F(TranscoderTest, JBossMarshalledCacheWriteIntegerJsonReadJBoss) {
```

Comment:
Can you add a comment here describing the purpose of this test as well?

Reply:
done